metaseq/cli/inference.py

```diff
@@ -0,0 +1,764 @@
+import argparse
+import json
+import logging
+import os
+import pathlib
+import shutil
+import sys
+import time
+import operator
+from collections import defaultdict
+from concurrent.futures import ThreadPoolExecutor
+from functools import partial
+from glob import glob
+from typing import Callable, Dict, List, Optional
+
+import torch
+from omegaconf import OmegaConf
+from tqdm import tqdm
+
+from metaseq import options, utils
+from metaseq.data import JsonlDataset, data_utils
+from metaseq.data.prompt_generator import PromptGenerator
+from metaseq.dataclass.configs import MetaseqConfig
+from metaseq.dataclass.utils import convert_namespace_to_omegaconf
+from metaseq.distributed import utils as distributed_utils
+from metaseq.generation_metrics.metrics import GenerationMetrics, evaluate_inference_files
+from metaseq.hub_utils import GeneratorInterface
+from metaseq.logging import get_logger
+from metaseq.utils import flatten_config
+
+logger = get_logger(__name__)
+
+TOTAL_TOKENS_GENERATED = 0
+
+
+def _tokenize_one_json(
+    json: Dict[str, str],
+    encode_fn: Callable[[str], List[int]],
+    prompt_generator: PromptGenerator,
+    prompt_eos_text: str,
+    delimiter=None,
+    max_seq_len: int = 2048,
+    max_prompt_len: int = 0,
+    truncation: str = "right",
+    prompt_end_text: Optional[str] = None,
+):
+    """
+    Inference Tokenization Function for Json Input
+
+    :param Dict[str, str] json: The json line containing the prompt and target
+    :param Callable encode_fn: The function to use to encode the prompt and
+        target. Encodes a string to a list of token ids.
+    :param PromptGenerator prompt_generator: The prompt generator to use
+    :param str prompt_eos_text: The text to append at the end of prompt
+    :param str delimiter: The delimiter to use to split the prompt and target
+    :param int max_seq_len: The maximum sequence length
+    :param int max_prompt_len: The maximum prompt length
+    :param Optional[str] prompt_end_text: If provided and the prompt
+        doesn't end with the specified text before being given to the model then
+        the last tokens of the prompt will be replaced with the tokens
+        corresponding to this text, defaults to None
+
+    :return: (torch.LongTensor, str) Prompt tokens and Target text
+    """
+    source_key = "text" if "text" in json else "src"
+    assert source_key in json, "json must contain a 'text' or 'src' field"
+
+    target_key = "tgt" if "tgt" in json else None
+
+    # NOTE: if you pass in a delimiter, we assume the target is already included
+    # in the `text/src` field, and we split on the delimiter to get the prompt
+    # and target respectively. That means that if you pass in a delimiter, you
+    # should not pass in a `tgt` field since it will be ignored.
```

Comment:
It will actually raise an `AssertionError`

```suggestion
    # should not pass in a `tgt` field.
```

or 

```suggestion
    # should not pass in a `tgt` field since it would be ignored.
```

Reply:
Hmm, I chose the top option of removal